AreaLight3D in Compatibility results in a shader compilation error when it affects a material using vertex shading

### Tested versions

- Reproducible in: 4.7.dev 90113707f

### System information

Godot v4.7.dev (90113707f) - Fedora Linux 43 (KDE Plasma Desktop Edition) on X11 - X11 display driver, Multi-window, 1 monitor - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 5090 (nvidia; 580.126.18) - AMD Ryzen 9 9950X3D 16-Core Processor (32 threads) - 62.33 GiB memory

### Issue description

[AreaLight3D](https://github.com/godotengine/godot/pull/108219) in Compatibility results in a shader compilation error when it affects a material using vertex shading:

```
ERROR: SceneShaderGLES3: Vertex shader compilation failed:
0(987) : error C1503: undefined variable "area_light_count"
0(988) : error C1503: undefined variable "light_process_area"
0(988) : error C1503: undefined variable "area_light_indices"

   at: _display_error_with_code (drivers/gles3/shader_gles3.cpp:262)
ERROR: Method/function failed.
   at: _compile_specialization (drivers/gles3/shader_gles3.cpp:356)
WARNING: shader failed to compile, unable to bind shader.
     at: _version_bind_shader (./drivers/gles3/shader_gles3.h:215)
```

This does not occur in Forward+/Mobile, where the area light is skipped on the vertex-shaded material (as vertex shading does not support area lights).

cc @CookieBadger 

Compatibility | Forward+
-|-
<img width="1152" height="648" alt="Image" src="https://github.com/user-attachments/assets/b31e64ed-cd5b-467b-a8b6-ea2f2070c8ae" /> | <img width="1152" height="648" alt="Image" src="https://github.com/user-attachments/assets/f6640488-7e7c-41eb-a1c3-9658e4e0ee78" />

### Steps to reproduce

- Use the Compatibility renderer.
- Add a MeshInstance3D node with a PlaneMesh.
- Add an AreaLight3D node lighting up the PlaneMesh.
- Assign a material to the MeshInstance3D mode, and set **Shading** to **Per-Vertex** on the material.

### Minimal reproduction project (MRP)

[test_area_light_vertex.zip](https://github.com/user-attachments/files/26728880/test_area_light_vertex.zip)

--- a/drivers/gles3/shaders/scene.glsl
+++ b/drivers/gles3/shaders/scene.glsl
@@ -878,13 +878,6 @@ void vertex_shader(vec4 vertex_angle_attrib_input,
 				diffuse_light_interp.rgb, specular_light_interp.rgb);
 	}
 #endif // !DISABLE_LIGHT_SPOT
-
-#ifndef DISABLE_LIGHT_AREA
-	for (uint i = 0u; i < area_light_count; i++) {
-		light_process_area(area_light_indices[i], vertex_interp, view, normal_interp, roughness,
-				diffuse_light_interp.rgb, specular_light_interp.rgb);
-	}
-#endif // !DISABLE_LIGHT_AREA
 #endif // BASE_PASS
 
 /* ADDITIVE LIGHTING PASS */

